lib/diaspora/federation/entities.rb

```diff
@@ -200,37 +177,29 @@ def self.reshare(reshare)
         )
       end
 
-      def self.retraction(target)
+      def self.retraction(retraction)
+        case retraction.data[:target_type]
+        when "Contact"
```

Comment:
And yes, this specific change (use `Contact` for stop sharing) doesn't depend on the gem, but it depends on other changes in this PR and other changes here depend on this change. So I'd like to keep it here if possible, because I think there would be many conflicts otherwise. Also I mixed it with the "use `Contact` for start sharing" change (which is the same issue in the federation repo, so I didn't split that to different commits), and the start sharing does depend on the gem, because I blocked the old `Request` entity in the gem.